srv/apns/http_api/processor_test.go

```diff
@@ -0,0 +1,130 @@
+package http_api
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+	"github.com/uniqush/uniqush-push/push"
+	"github.com/uniqush/uniqush-push/srv/apns/common"
+)
+
+const (
+	authToken = "test_auth_token"
+	keyFile   = "../apns-test/localhost.p8"
+	keyID     = "FD8789SD9"
+	teamID    = "JVNS20943"
+	bundleID  = "com.example.test"
+)
+
+var (
+	pushServiceProvider = &push.PushServiceProvider{
+		push.PushPeer{
+			VolatileData: map[string]string{
+				"addr": "https://api.development.push.apple.com",
+			},
+			FixedData: map[string]string{
+				"p8":       keyFile,
+				"keyid":    keyID,
+				"teamid":   teamID,
+				"bundleid": bundleID,
+			},
+		},
+	}
+	devToken = []byte("test_device_token")
+	payload  = []byte(`{"alert":"test_message"}`)
+	apiURL   = fmt.Sprintf("%s/3/device/%s", pushServiceProvider.VolatileData["addr"], hex.EncodeToString(devToken))
+)
+
+type MockJWTManager struct{}
+
+func (*MockJWTManager) GenerateToken() (string, error) {
+	return authToken, nil
+}
+
+func mockAPNSRequest(fn func(r *http.Request) (*http.Response, error)) {
+	httpmock.RegisterResponder("POST", apiURL, fn)
+}
+
+func newPushRequest() (*common.PushRequest, chan push.PushError, chan *common.APNSResult) {
+	errChan := make(chan push.PushError)
+	resChan := make(chan *common.APNSResult, 1)
+	request := &common.PushRequest{
+		PSP:       pushServiceProvider,
+		Devtokens: [][]byte{devToken},
+		Payload:   payload,
+		ErrChan:   errChan,
+		ResChan:   resChan,
+	}
+
+	return request, errChan, resChan
+}
+
+var mockJWTManager = &MockJWTManager{}
+
+func TestAddRequestPushSuccessful(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	request, _, resChan := newPushRequest()
+	mockAPNSRequest(func(r *http.Request) (*http.Response, error) {
+		// Return empty body
+		return httpmock.NewBytesResponse(http.StatusOK, nil), nil
```

Comment:
of the **request** made by Uniqush (`r *http.Request`).

- i.e. of  r.Headers and ioutil.readAll(r.Body) (I forgot the exact names)